### PR TITLE
Add update_data.py script

### PR DESCRIPTION
**Pull request by dcolligan**
Sample run:
```
$ python scripts/update_data.py
Archiving 'ga4gh-example-data' to 'ga4gh-example-data-archive/Wed_Feb_25_13:06:03_2015/ga4gh-example-data'
Archiving 'ga4gh-example-data.tar.gz' to 'ga4gh-example-data-archive/Wed_Feb_25_13:06:03_2015/ga4gh-example-data.tar.gz'
Archiving finished
Downloading 'http://www.well.ox.ac.uk/~jk/ga4gh-example-data.tar.gz' to 'ga4gh-example-data.tar.gz'
Downloading... 100.00% finished (  164M / 164M  )  # interactive ftw
Downloading finished
Extracting ga4gh-example-data.tar.gz
Extracting finished
Finished in 1 minute(s), 59 second(s)
```

Issue #189

### CODE REVIEW

**Comment by dcolligan** [comment]
You know, we may just be able to include this in the `.travis.yml` if we wanted to run end-to-end tests on travis... not sure if that's either doable or desirable.

**Comment by pgrosu** [comment]
+1, looks great!  I agree that having it in travis would help consistency and completeness, as long as the data is not too large. For now probably not having it in travis would speed things up, since I'm sure we might go through other iterations of how the data will get structured and thus downloaded.

Many thanks for creating this,
Paul

**Comment by dcolligan** [comment]
Added an update... turns out the humanize module has got tons of functionality that I basically duplicated in a previous commit.

File: scripts/update_data.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            return result
+        return wrapper
+
+    def _report(self):
+        delta = self.end - self.start
+        timeString = humanize.time.naturaldelta(delta)
+        log("Finished in {} ({} seconds)".format(timeString, delta))
+
+
+class FileDownloader(object):
```
Comment (dcolligan) [inline scripts/update_data.py]: should this class be moved to its own module?  if so, where?

Comment (jeromekelleher) [inline scripts/update_data.py]: We do something very similar in `process_schemas.py`. Maybe we could have a `scripts/utils.py` module and reuse the FileDownloader here and in `process_schemas`? We'll need to start worrying about where these things are run, but it should be workable.

**Comment by jeromekelleher** [comment]
This looks great @dcolligan, thanks. What do you think of the `scripts/utils.py` module? Perhaps this would also be a natural home for the decorators you proposed also in #192?

File: scripts/update_data.py (hunk 1 above)
Comment (dcolligan) [inline scripts/update_data.py]: `scripts/utils.py` would be a good idea and I agree that `FileDownloader` should go in there.

I think we can safely assume that everything in `scripts` is run from the root server directory.

**Comment by dcolligan** [comment]
The decorators should go in their own module under `ga4gh` since they are so general purpose and right now used mostly in `tests`.  I could see them being used in core server code as well.

**Comment by Naburimannu** [comment]
+1, just holding off the merge because it isn't clear that Jerome & Danny have reached agreement?

**Comment by dcolligan** [comment]
Yeah, hold off on the merge; I'm doing some refactoring.

**Comment by jeromekelleher** [comment]
It's up to Danny - I'm happy to merge on this if he'd rather do the `scripts/utils` thing later (or not at all).

**Comment by dcolligan** [comment]
Alright, pushed an update.  (Note the new launching syntax detailed in the commit message)

File: scripts/google_examples.py [hunk 2]
```diff
@@ -11,6 +10,8 @@
 
 import yaml
 
+import scripts.utils as utils
```
Comment (jeromekelleher) [inline scripts/google_examples.py]: If we express this as 
```python
import utils
```
we don't need to change the way we run the scripts. We also don't need to turn `scripts` into a package, so there's no need for the `__init__.py` file.

**Comment by jeromekelleher** [comment]
This is really cool, thanks @dcolligan. Should we change the import structure? I think it'd be a bit simpler to use the `import utils` form, and would mean we can continue using `python scripts/x.py` method to run things, which means we can tab-complete in the shell.

Other than that, a resounding +1.

**Comment by Naburimannu** [comment]
I certainly prefer doing it the way @jeromekelleher suggests, but I thought
@dcolligan said that wasn't working for him for some reason? Did we figure
out why?

**Comment by dcolligan** [comment]
Alright, reverted to the old script launch method... should be good to merge now.

**Pull request merged**